internal/common/voting/election_test.go

```diff
@@ -0,0 +1,35 @@
+package voting
+
+import (
+	"testing"
+
+	"github.com/SOMAS2020/SOMAS2020/internal/common/baseclient"
+	"github.com/SOMAS2020/SOMAS2020/internal/common/shared"
+)
+
+func TestElection(t *testing.T) {
+	var ele Election
+	ele.roleToElect = 0
+	ele.votingMethod = 0
+	ele.candidateList = []shared.ClientID{shared.Team1, shared.Team2, shared.Team3, shared.Team4, shared.Team5, shared.Team6}
+	ele.voterList = ele.candidateList
+	ele.votes = [][]shared.ClientID{{shared.Team3, shared.Team2, shared.Team1, shared.Team6, shared.Team5, shared.Team4},
+		{shared.Team4, shared.Team6, shared.Team5, shared.Team2, shared.Team3, shared.Team1},
+		{shared.Team3, shared.Team6, shared.Team1, shared.Team4, shared.Team5, shared.Team2},
+		{shared.Team2, shared.Team5, shared.Team6, shared.Team4, shared.Team3, shared.Team1},
+		{shared.Team6, shared.Team4, shared.Team1, shared.Team5, shared.Team2, shared.Team3},
+		{shared.Team5, shared.Team2, shared.Team3, shared.Team6, shared.Team1, shared.Team4}}
+	clientMap := make(map[shared.ClientID]baseclient.Client)
+
+	var c baseclient.BaseClient
+
+	for i := 0; i < len(ele.voterList); i++ {
+		clientMap[ele.voterList[i]] = &c
+	}
+
+	ele.bordaCountResult()
+	ele.runOffResult(clientMap)
+	ele.instantRunoffResult(clientMap)
+	ele.approvalResult()
```

Comment:
are you checking the results

Reply:
yeah, it would be hard to determine who is the winner in advance, as preference list is randomly set in each literation, unless observing information generated by each literation to see whether it works properly or not like what I manually tested before I think? : (

Reply:
hmm... you should make the place where you use random things testable. You can instead inject the required shuffled list into the function to test.

For example, say I want a sum of randomly selected numbers in a list: I can do
```
func getRandomSumFromList(lst []int, randIndices []int) int {
    sum := 0
    for _, i := range randIndices {
        // usually we also check for lst bounds here
        sum += lst[i]
    }
    return sum
}
```
In your test you can then make some `randIndices` to test your function. This function is therefore _pure_ -- it does not rely on anything else other than its inputs.

When using this function in an area you need randomness, you get that in the calling function then pass this in here.

Reply:
You can also use _dependency injection_ to inject a function returning randIndices into the function. This is the best approach.

I won't force you to do this here, but if you can guarantee that this is working properly to the primary users (IIGO @Neelesh99  @preetl), I'm fine with it

Reply:
I'll try it soon, thanks!

Reply:
I rewrite functions in test file, setting more return values needed for testing, is that exactly what you mean?